File: 137-single-number-ii/single-number-ii.cpp
class Solution {
public:
    int singleNumber(vector<int>& arr) {
        int n=(int) arr.size();
        vector<int> bitt(32);
        for(int i=0;i<n;++i){
            for(int j=0;j<32;++j){
                if((1<<j)&arr[i]) bitt[j]++;
            }
        }
        int ans=0;
        for(int j=0;j<32;++j){
            bitt[j]%=3;
            if(bitt[j]) ans+=(1<<j);
        }
        return ans;
    }
};

File: 137-single-number-ii/single-number-ii.java
class Solution {
    private boolean checkBit(int n, int k){
        return (n&(1<<k))!=0;
    }
    public int singleNumber(int[] nums) {
        int ans= 0;
        for(int i=0;i<32;i++){
            int count= 0;
            for(int x: nums){
                if(checkBit(x, i)) count++;
            }
            if(count%3==1) ans+= 1<<i;
        }
        return ans;
    }
}